docs/config.md

```diff
@@ -663,11 +663,23 @@ java -Dspring.config.location=file:/path/to/your/myconfig.yaml -jar promregator-
 #### Option "promregator.authenticator.oauth2xsuaa.tokenServiceURL" (mandatory, if using promregator.authenticator.type=OAuth2XSUAA)
 Specifies the URL of the OAuth2 endpoint, which contains the token service of your authorization server in case of global authentication. Typically, this is the endpoint with the path `/oauth/token`, as Promregator will try to perform to establish a ["Client Credentials"-based authentication](https://www.digitalocean.com/community/tutorials/an-introduction-to-oauth-2#grant-type-client-credentials).
 
+#### Option "promregator.authenticator.oauth2xsuaa.url"
+Like `promregator.authenticator.oauth2xsuaa.tokenServiceURL`, but without trailing `/oauth/token`.
+
+#### Option "promregator.authenticator.oauth2xsuaa.cert_url"
+Specifies the URL of the OAuth2 endpoint for certificate based authentication, which contains the token service of your authorization server in case of global authentication. Needs to be provided without a path.
+
 #### Option "promregator.authenticator.oauth2xsuaa.client_id" (mandatory, if using promregator.authenticator.type=OAuth2XSUAA)
 Specifies the client identifier (a.k.a. "client_id") which shall be used during the OAuth2 request based on the Grant Type Client Credentials flow in case of global authentication.
 
-#### Option "promregator.authenticator.oauth2xsuaa.client_secret" (mandatory, if using promregator.authenticator.type=OAuth2XSUAA)
-Specifies the client secret (a.k.a. "client_secret") which shall be used during the OAuth2 request based on the Grant Type Client Credentials flow in case of global authentication.
+#### Option "promregator.authenticator.oauth2xsuaa.client_secret" (mandatory for client secret based authentication, if using promregator.authenticator.type=OAuth2XSUAA)
+Specifies the client secret (a.k.a. "client_secret") which shall be used during the OAuth2 request based on the Grant Type Client Credentials flow in case of global authentication. Omitted when certificate based authentication is used.
+
+#### Option "client_certificates" (mandatory for certificate based authentication, if using promregator.authenticator.type=OAuth2XSUA)
```

Comment:
Not sure about the naming. I used the plural here `client_certificates` since it is a certificate chain. Of course only the last element in the chain is really important. Would also be possible to use the singular `client_certificate` here.

Reply:
We also have other cases here in the configuration which uses the plural form, e.g. `promregator.scraping.threads`  and `.scopes` although that is a single value (and not a YAML array). I'd say that we use the most common and best understandable term to express "what's behind that attribute". I assume that it the vast majority of cases, we have more than one certificate in there (due to the intermediate certificate). So, if that's true, I think it should be called `client_certificates` (plural).

BTW: We should briefly mention the separator approach between different values. I assume it's "concatenated string, no separation necessary, as the certificates have a suffix and prefix".
Are users allowed to use newlines (escaping them?) or is this prohibited?

Reply:
> Are users allowed to use newlines (escaping them?) or is this prohibited?

The users should simply use what is provided in the service key files. There we have a long string containing all the certificates with the normal `-----BEGIN CERTIFICATE-----\n` and `-----END CERTIFICATE-----\n` The next `-----BEGIN CERTIFICATE-----\n` is right after `-----END CERTIFICATE-----\n`. I looks like: `...L5MSJYw==\n-----END CERTIFICATE-----\n-----BEGIN CERTIFICATE-----\nMIIGY...`

Of course we have also `\n` inbetween as usual.